### PR TITLE
Figs 1&2 Code

### PR DESCRIPTION
**Pull request by murphymadeleine21**
Degrees of freedom vs. variance figure and heatmap figure code.

### CODE REVIEW

**Review by cyrillustan — COMMENTED** [review]
Looks great! Maybe some cosmetic changes to adjust the size for figures and use integers for day numbers would make them more good-looking.

File: syserol/figures/figure2.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    days = dayLabels()
+    tfac = CMTFfacs[1]
+
+    df = pbsSubtractOriginal()
+    components = [str(ii + 1) for ii in range(tfac.rank)]
+    comp_plot(tfac.factors[0], components,
+              list(df.loc[np.unique(df['patient_ID'])]['group']), "Subjects", ax[0], True)
+    comp_plot(tfac.factors[1], components, agLabels, "Antigens", ax[1])
+    comp_plot(tfac.factors[2], components, Rlabels, "Receptors", ax[2])
+    comp_plot(tfac.factors[3], components, days, "Time (days)", ax[3])
```
Comment (aarmey) [inline syserol/figures/figure2.py:33]: Don't have to do it on this PR, but I think the time would be visualized more effectively as a line plot. For example, see Fig. 2F here: https://www.biorxiv.org/content/10.1101/2021.07.03.451002v1

Comment (murphymadeleine21) [reply syserol/figures/figure2.py]: So in that figure, some of them don't look exactly unimodal? For example, component 1 goes up and then back down?

Comment (aarmey) [reply syserol/figures/figure2.py]: Right. That's not using your code. The visualization is still useful as an example.

Comment (murphymadeleine21) [reply syserol/figures/figure2.py]: I see, I thought maybe because it was also time it would also have similar behavior.

**Comment by murphymadeleine21** [comment]
Might just merge this as it is working for now, and do any other changes later?

**Review by cyrillustan — APPROVED** [review]
Yes, I agree with Dr. Meyer that representing the time mode with a linear plot makes more sense (and we can compare them with the results of our new method). We don't have to do it in this PR though. Let's merge this now.

**Pull request merged**